sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/util/TimestampFormatterSuite.scala

```diff
@@ -502,4 +504,25 @@ class TimestampFormatterSuite extends DatetimeFormatterSuite {
     assert(formatter.parseOptional("9999-12-31 23:59:59.999").isEmpty)
     assert(formatter.parseWithoutTimeZoneOptional("9999-12-31 23:59:59.999", true).isEmpty)
   }
+
+  test("fail to parse string as TimestampNTZ with invalid format") {
+    val zoneId = ZoneId.systemDefault()
+    val locale = Locale.getDefault()
+    val formatter = new DefaultTimestampFormatter(
+      zoneId, locale, LENIENT_SIMPLE_DATE_FORMAT, isParsing = true)
+
+    val invalidTimestampStr = "2021-13-01T25:61:61"
+
+    checkError(
+      exception = intercept[SparkRuntimeException] {
+        formatter.parseWithoutTimeZone(invalidTimestampStr, allowTimeZone = false)
```

Comment:
Could you reproduce the error using public API, please. Just ask you because if we cannot do that, we should consider to convert it to an internal error.

Reply:
I just realized the comment from the source code below:

```
* The formatter for timestamps which doesn't require users to specify a pattern. While formatting,
 * it uses the default pattern [[TimestampFormatter.defaultPattern()]]. In parsing, it follows
 * the CAST logic in conversion of strings to Catalyst's TimestampType.
```

so I believe this cannot be reproduced from user space, so we should mark it as an internal error.